### ISSUE TITLE
Missing support for adding certificate chains to the hypervisor memory

### ISSUE THREAD

**Issue by ngiraud99** [virtee/snphost#111]
Hi,

I'm currently looking for an option to add a certificate-chain to the hypervisor memory to be able to perform the extended attestation workflow on an SNP host.
Is there already a method in this project to do this?
I already had a look at the respective [AMD Documentation](https://www.amd.com/content/dam/amd/en/documents/developer/58217-epyc-9004-ug-platform-attestation-using-virtee-snp.pdf) and it seems that this feature is supported by AMD under section 2.2.2.

Thanks in advance!

**Comment by tylerfanelli** [comment]
Yes, it's supported, but is hypervisor-specific. I believe the documentation you're pointing to is showing how hypervisors importing this crate can easily store certificate chains in their memory using the API.

What hypervisor are you using?

**Comment by tylerfanelli** [comment]
Closed this as =

**Comment by tylerfanelli** [comment]
Closing as its not a snphost bug/feature.

**Comment by DGonzalezVillal** [comment]
@aisuneko Since the functionality was not approved by the upstream kernel community, the functionality was never added to upstream kernel and QEMU. I think they also stripped the functionality from the development branches of the SNP kernel and QEMU. So as far as I know, unless it is with an old devel kernel and qemu, there is currently no way of getting those extended certificates. You can try and build the devel kernel in https://github.com/AMDESE/AMDSEV/tree/snp-latest and try, but no guarantee this will work right now.

Inside the guest, the command to get the imported certs is `snpguest certifcates`.

Once there is an approved and up streamed workflow, we will update the tooling to be able to get extended certs.
